sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_client_async.py

```diff
@@ -55,68 +52,68 @@
-     or a receive source.
-    :type remote_address: str, bytes or ~uamqp.address.Address
-    :param auth: Authentication for the connection. This should be one of the subclasses of
-     uamqp.authentication.AMQPAuth. Currently this includes:
-        - uamqp.authentication.SASLAnonymous
-        - uamqp.authentication.SASLPlain
-        - uamqp.authentication.SASTokenAsync
+    :param hostname: The AMQP endpoint to connect to.
+    :type hostname: str
+    :keyword auth: Authentication for the connection. This should be one of the following:
+        - pyamqp.authentication.SASLAnonymous
+        - pyamqp.authentication.SASLPlain
+        - pyamqp.authentication.SASTokenAuth
+        - pyamqp.authentication.JWTTokenAuth
      If no authentication is supplied, SASLAnnoymous will be used by default.
-    :type auth: ~uamqp.authentication.common.AMQPAuth
-    :param client_name: The name for the client, also known as the Container ID.
+    :paramtype auth: ~pyamqp.authentication
+    :keyword client_name: The name for the client, also known as the Container ID.
      If no name is provided, a random GUID will be used.
-    :type client_name: str or bytes
-    :param loop: A user specified event loop.
-    :type loop: ~asycnio.AbstractEventLoop
-    :param debug: Whether to turn on network trace logs. If `True`, trace logs
+    :paramtype client_name: str or bytes
+    :keyword debug: Whether to turn on network trace logs. If `True`, trace logs
      will be logged at INFO level. Default is `False`.
-    :type debug: bool
-    :param error_policy: A policy for parsing errors on link, connection and message
+    :paramtype debug: bool
+    :keyword retry_policy: A policy for parsing errors on link, connection and message
      disposition to determine whether the error should be retryable.
-    :type error_policy: ~uamqp.errors.ErrorPolicy
-    :param keep_alive_interval: If set, a thread will be started to keep the connection
+    :paramtype retry_policy: ~pyamqp.error.RetryPolicy
+    :keyword keep_alive_interval: If set, a thread will be started to keep the connection
      alive during periods of user inactivity. The value will determine how long the
      thread will sleep (in seconds) between pinging the connection. If 0 or None, no
      thread will be started.
-    :type keep_alive_interval: int
-    :param max_frame_size: Maximum AMQP frame size. Default is 63488 bytes.
-    :type max_frame_size: int
-    :param channel_max: Maximum number of Session channels in the Connection.
-    :type channel_max: int
-    :param idle_timeout: Timeout in seconds after which the Connection will close
+    :paramtype keep_alive_interval: int
+    :keyword max_frame_size: Maximum AMQP frame size. Default is 63488 bytes.
+    :paramtype max_frame_size: int
+    :keyword channel_max: Maximum number of Session channels in the Connection.
+    :paramtype channel_max: int
+    :keyword idle_timeout: Timeout in seconds after which the Connection will close
```

Comment:
Does idle_timeout actually do anything?

Reply:
It gets passed through to Connection, then to OpenFrame to set service timeout on idle connection, if not set though it doesn't impact anything. Default is None